packages/react/src/components/Tabs/Tabs.js

```diff
@@ -133,38 +135,52 @@ export default class Tabs extends React.Component {
     };
   };
 
-  handleTabKeyDown = onSelectionChange => {
-    return (index, evt) => {
-      const key = evt.key || evt.which;
-
-      if (key === 'Enter' || key === 13 || key === ' ' || key === 32) {
-        this.selectTabAt(index, onSelectionChange);
-        this.setState({
-          dropdownHidden: true,
-        });
-      }
-    };
+  handleTabKeyDown = onSelectionChange => (index, evt) => {
+    if (matches(evt, [keys.Enter, keys.Space])) {
+      this.selectTabAt(index, onSelectionChange);
+      this.setState({
+        dropdownHidden: true,
+      });
+    }
   };
 
-  handleTabAnchorFocus = onSelectionChange => {
-    return index => {
-      const tabCount = React.Children.count(this.props.children) - 1;
-      let tabIndex = index;
-      if (index < 0) {
-        tabIndex = tabCount;
-      } else if (index > tabCount) {
-        tabIndex = 0;
+  handleTabAnchorFocus = onSelectionChange => (evt, { index }) => {
+    const newIndex = (() => {
```

Comment:
Does this need to be wrapped in an IIFE?

Reply:
not really, but we immediately execute the function and assign the return value to `newIndex` this way

Reply:
@emyarod right, I'm just curious why we need to do this versus hoisting it to the parent scope?

Reply:
not sure I fully understand your question, but it references `index` so it needs to be within the scope of `handleTabAnchorFocus`